nav2_dwb_controller/dwb_critics/src/map_grid.cpp

```diff
@@ -117,9 +117,11 @@ double MapGridCritic::scoreTrajectory(const dwb_msgs::msg::Trajectory2D & traj)
     grid_dist = scorePose(traj.poses[i]);
     if (stop_on_failure_) {
       if (grid_dist == obstacle_score_) {
-        throw nav_core2::IllegalTrajectoryException(name_, "Trajectory Hits Obstacle.");
+        throw nav2_local_planner_interface::
```

Comment:
I agree.
Do you think this package name should reflect whether it has local planner or global planner core in it? Or should this have both global and local planner interfaces in it in the future? If it will contain both the interfaces for global and local planning, we can name it as 'nav2_core'.
During my conversation with the team here (Intel), we thought of keeping global and local planners separate makes sense. However, 'nav2_local_planner_interfaces' and 'nav2_global_planner_interfaces' would be too long names. Maybe 'nav2_local_core', and 'nav2_global_core' for the interfaces?

Reply:
I’ll challenge the assumption of the question ;-) I DON’T think it should be named to reflect local or global, because it should contain both! I’d encourage you to take a look at a bunch of the _core packages at exist in the community, especially ones that are in metapackages. 

There’s power in consistency, and do you really want to look at a project and see 35 packages, 10 of which contain <4 headers? :-)

Reply:
So more context-
The reason the interfaces are in another package are so that you don’t have to install an implementation to get the abstract header to make your new shiny plugin. However installing 4 vs 10 header-only files isnt installing any executables to get your include headers to implement a thing. Splitting them _too_ much is a bit of a slippery slope. A similar analog would be: well why does nav2_utils contain costmap and amcl utilities? Why not have nav2_amcl_utils and nav2_costmap_2d_utils- then a thousand utility packages.